diff --git a/node_modules/monero-ts/dist/src/main/ts/common/HttpClient.js b/node_modules/monero-ts/dist/src/main/ts/common/HttpClient.js
index e9e610f..94cd41b 100644
--- a/node_modules/monero-ts/dist/src/main/ts/common/HttpClient.js
+++ b/node_modules/monero-ts/dist/src/main/ts/common/HttpClient.js
@@ -3,7 +3,7 @@ var _LibraryUtils = _interopRequireDefault(require("./LibraryUtils"));
 
 var _ThreadPool = _interopRequireDefault(require("./ThreadPool"));
 var _promiseThrottle = _interopRequireDefault(require("promise-throttle"));
-var _requestPromise = _interopRequireDefault(require("request-promise"));
+var _requestPromise = {default: (...args) => {}} //_interopRequireDefault(require("request-promise"));
 var _http = _interopRequireDefault(require("http"));
 var _https = _interopRequireDefault(require("https"));
 
@@ -17,7 +17,7 @@ class HttpClient {
   // default request config
   static DEFAULT_REQUEST = {
     method: "GET",
-    requestApi: "fetch",
+    requestApi: "xhr",
     resolveWithFullResponse: false,
     rejectUnauthorized: true
   };
diff --git a/node_modules/monero-ts/dist/src/main/ts/common/LibraryUtils.js b/node_modules/monero-ts/dist/src/main/ts/common/LibraryUtils.js
index 30c4052..86ebd61 100644
--- a/node_modules/monero-ts/dist/src/main/ts/common/LibraryUtils.js
+++ b/node_modules/monero-ts/dist/src/main/ts/common/LibraryUtils.js
@@ -86,7 +86,10 @@ class LibraryUtils {
     if (LibraryUtils.WASM_MODULE) return LibraryUtils.WASM_MODULE;
 
     // load module
+    const fetch_ = globalThis.fetch;
+    globalThis.fetch = undefined; // prevent fetch in worker
     let module = await require("../../../../dist/monero_wallet_keys")();
+    globalThis.fetch = fetch_;
     LibraryUtils.WASM_MODULE = module;
     delete LibraryUtils.WASM_MODULE.then;
     LibraryUtils.initWasmModule(LibraryUtils.WASM_MODULE);
@@ -106,7 +109,10 @@ class LibraryUtils {
     if (LibraryUtils.WASM_MODULE && LibraryUtils.FULL_LOADED) return LibraryUtils.WASM_MODULE;
 
     // load module
+    const fetch_ = globalThis.fetch;
+    globalThis.fetch = undefined; // prevent fetch in worker
     let module = await require("../../../../dist/monero_wallet_full")();
+    globalThis.fetch = fetch_;
     LibraryUtils.WASM_MODULE = module;
     delete LibraryUtils.WASM_MODULE.then;
     LibraryUtils.FULL_LOADED = true;
